easybuild/easyconfigs/a/AOMP/AOMP-11.12-0-gcccuda-2019b.eb

```diff
@@ -0,0 +1,63 @@
+easyblock = 'Binary'
+
+name = 'AOMP'
+version = '11.12-0'
+
+homepage = 'https://github.com/ROCm-Developer-Tools/aomp'
+description = "AMD fork of LLVM, setup for OpenMP offloading to Accelerators"
+
+toolchain = {'name': 'gcccuda', 'version': '2019b'}
+
+source_urls = ['https://github.com/ROCm-Developer-Tools/aomp/releases/download/rel_%(version)s/']
+sources = ['aomp-%(version)s.tar.gz']
+patches = ['build_aomp-%(version)s.patch']
+checksums = [
+    'b01bbed55c4851cfebc69d5be42bafdfb8b43b82e0b6715fe5e1e86db47272e2',  # aomp-11.12-0.tar.gz
+    'c5ccdd00e270731a353461a8f158cdec556689742d501163c8059e739d0588b7',  # build_aomp-11.12-0.patch
+]
+
+builddependencies = [
+    ('CMake', '3.15.3'),
+    ('Perl', '5.30.0'),
+    ('Python', '3.7.4'),
+    ('pkg-config', '0.29.2'),
+]
+
+dependencies = [
+    ('binutils', '2.32'),
+    ('libffi', '3.2.1'),
+    ('numactl', '2.0.12'),
+]
+
+extract_sources = True
+dontcreateinstalldir = True
+
+preinstallopts = 'AOMP=%(installdir)s '
+preinstallopts += 'AOMP_REPOS="%(builddir)s/aomp%(version_major)s" '
+preinstallopts += 'AOMP_CHECK_GIT_BRANCH=0 '
+preinstallopts += 'AOMP_APPLY_ROCM_PATCHES=0 '
+preinstallopts += 'AOMP_STANDALONE_BUILD=1 '
+preinstallopts += 'NUM_THREADS=%(parallel)s'
+# 'NVPTXGPUS' should be set to the NVPTX architecture you would like to
+# support, e.g. '50,75'.
+preinstallopts += ' AOMP_BUILD_CUDA=1 '
+preinstallopts += 'CUDA="$CUDA_ROOT" '
+preinstallopts += 'NVPTXGPUS="70"'
```

Comment:
I think this should be configurable. Does it support 7.0 or sm_70 format? Then you can use the templates cuda_compute_capabilities etc.

Reply:
Based on the [pre-configured](https://github.com/ROCm-Developer-Tools/aomp/blob/master/bin/aomp_common_vars#L110) targets, I assumed one had to use whole numbers only.

Reply:
Hm, then I'd say an EasyBlock is better here. I'm afraid short of patching the EC (e.g. via hooks) this would limit the GPU usability unusually, especially after we introduced the `--cuda-compute-capabilities` option
This has the additional benefit that those hard-coded, repeated preinstallopts can be unified and handled better (e.g. detecting presence of CUDA)